WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
> Redefining a core framework method like find scares me. I don't know who might be calling it and expecting the regular functionality.

Argh yes, I even read this from your previous comment and it slipped through my mind...

I totally agree that the class is hard to understand at first sight, I added a comment of why it is like this!
However I would not like to do code duplication just because the code is hard to read, I'd rather detail the base class implementation.
Here what we want to achieve (same behavior shared by multiple classes) is perfectly solved by inheriting an abstract class, so it feels hard for me to say no to that just because the base class is not trivial :s